src/main/java/rape/brutal/gitstarship/Storage.java

```diff
@@ -36,11 +36,16 @@ public void setCapacity(int capacity) {
         return itemsHashMap.keySet();
     }
 
-    public void putItem(IStorageable iStorageable) {
+    public boolean putItem(IStorageable iStorageable) {
 
+        if ((capacity - iStorageable.getSize()) >= 0) {
```

Comment:
It is only architecture fix, not feature fix. There is should be no realization.